core/app/models/spree/payment_method.rb

```diff
@@ -29,13 +27,12 @@ def payment_source_class
     def self.available(display_on = 'both')
       all.select do |p|
```

Comment:
I am not really aiming to change any of the logic or structure with this PR, but I have a plan to tackle the display_on madness next.